plugin/evm/uptime/pausable_manager.go

```diff
@@ -0,0 +1,146 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package uptime
+
+import (
+	"errors"
+
+	"github.com/ava-labs/subnet-evm/plugin/evm/validators"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/uptime"
+	"github.com/ava-labs/avalanchego/utils/set"
+)
+
+var _ validators.StateCallbackListener = &pausableManager{}
+
+var errPausedDisconnect = errors.New("paused node cannot be disconnected")
+
+type PausableManager interface {
```

Comment:
this is a recurring pattern I see in avalanche codebase - defining an interface where it's implemented, instead of where it's used. 

What is the benefit of doing it this way? 

Anyone that wants to use this interface needs to import this package directly, so you're creating coupling between any consumer package to this package.

Reply:
Yea I agree this not ideal. I think this is an approach that is valid until it breaks. Ideally we should have a separate subpackage for interfaces.

Reply:
Well I think ideally, the interface is defined where it's consumed, not where it's implemented.

This is not related to this PR, however, it's a general comment.

Reply:
https://github.com/ava-labs/subnet-evm/pull/1379/files